src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1576,6 +1576,11 @@ private List<ActivationData> TryGetActivationDatas(List<ActivationAddress> addre
 
         public void SiloStatusChangeNotification(SiloAddress updatedSilo, SiloStatus status)
         {
+            this.ScheduleTask(() => this.OnSiloStatusChange(updatedSilo, status));
```

Comment:
Task returned by `this.ScheduleTask()` is not awaited or `Ignore`'ed. I think it's best to wrap it in a `SafeExecute()`, so that potential exceptions are handled and logged. Alternatively, we could add another extension that would do that and return void instead of a `Task`.